### ISSUE TITLE
Demand generating taking too long to finish

### ISSUE THREAD

**Issue by jafshin** [matsim-melbourne/demand#3]
Here is the log file for a 5% sample, it is using the algorithm until 266d4811c8b25a3eff1094e74752401c4819765d:

[makeExamplePopulation.log](https://github.com/matsim-melbourne/demand/files/5310996/makeExamplePopulation.log)

In summary:
- Total run time: 9h and 8m
- Generating 100000 VISTA-like daily travel plans into ../output/3.plan/plan.csv: 35m
- Assigning home SA1 locations: 17m
- Assigning activities' SA1s and travel modes: 4h 42m
- Assigning coordinates to activities in SA1s: 1h 37m
- Assigning start/end times to activities: 1h 39m
- Writing as MATSim XML: 12m

Possibility of parallelization in some steps has been discussed as a solution.

**Comment by dhixsingh** [comment]
@alanboth if you could populate this issue with the manual steps currently required to run the speedup location assignment code.  We can then look at integrating that in to replace the slower existing algorithm.

**Comment by alanboth** [comment]
Sure thing, here's the code needed to run the parallel code:

```
# run setup through to match ----------------------------------------------
source("makeExamplePopulation.R")
samplePercent=0.1
numPlans=5000
# Run: 1. setup, 2. sample, 3. plan, and 4. match
makeExamplePopulation(samplePercent,numPlans,do.steps=c(T,T,T,T,F,F,F,F))

# parallel workaround -----------------------------------------------------
source("util.R")
source("locations.R")
loadLocationsData(
  '../output/1.setup/locDistanceMatrix.rds',
  '../output/1.setup/locDistanceMatrixIndex.rds',
  '../output/1.setup/locSa1Aattributed.rds',
  '../output/1.setup/locSa1Centroids.rds',
  '../output/1.setup/locAddresses.rds'
)
censuscsv='../output/2.sample/sample.csv.gz'
vistacsv='../output/3.plan/plan.csv'
matchcsv='../output/4.match/match.csv.gz'
outdir='../output/5.locate'
outcsv='../output/5.locate/plan.csv'
source('locateParallel.R')
```

**Comment by dhixsingh** [comment]
@alanboth, looks like you have changed the input files here:
https://github.com/matsim-melbourne/demand/blob/34ace980367852525d5b615f693e0b32af565c6b/R/makeExamplePopulation.R#L46-L53
The last three parameters used to be `.sqlite` but are now `.csv.gz`. If the inputs have changed, can you send me the new files so I can upload them to the data folder on Cloudstor and remove the outdated ones?

**Comment by dhixsingh** [comment]
@alanboth I've now integrated the parallel locate code into the algorithm, and have sorted out all the scoping issues, added a new test for the parallel code, and also updated the expectations so that all the test pass. 

A couple of issues remain.  The first I am tracking separately in #8. The second is that `makeExamplePopulation` now fails at the next step. To reproduce, do:
```
Rscript -e 'setwd("R"); source("makeExamplePopulation.R"); runexample()'
```

Which eventually fails at step 6:
```
2020-11-13 13:35:05 | Loading VISTA-like plans from ../output/5.locate/plan.csv
2020-11-13 13:35:05 | Assigning coordinates to activities in SA1s (can take a while)
2020-11-13 13:35:05 | ..........|..........|..........|..........|..........|..........|..........|..........|...Error in eval(jsub, SDenv, parent.frame()) : object 'x' not found
Calls: runexample ... %>% -> eval -> eval -> [ -> [.data.table -> eval -> eval
Execution halted
```

If you could take it from here and have a look at the above two issues.

**Comment by dhixsingh** [comment]
Fixed in 059be45.
